plugins/maxruntime/maxruntime.go

```diff
@@ -0,0 +1,88 @@
+package maxruntime
+
+import (
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	schematypes "github.com/taskcluster/go-schematypes"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/plugins"
+)
+
+type pluginProvider struct {
+	plugins.PluginProviderBase
+}
+
+type plugin struct {
+	plugins.PluginBase
+}
+
+type taskPlugin struct {
+	plugins.TaskPluginBase
+	maxRunTime int
+	done       chan bool
+	log        *logrus.Entry
+}
+
+type payloadType struct {
+	MaxRunTime int `json:"maxRunTime"`
+}
+
+var payloadSchema = schematypes.Object{
+	Properties: schematypes.Properties{
+		"maxRunTime": schematypes.Integer{
+			MetaData: schematypes.MetaData{
+				Title:       "Task execution timeout (ms)",
```

Comment:
Honestly, I'm tempted to want support for strings like `'5 hours'` and `'35 min'`, and then just defined a regexp for it. But that's probably crazy overkill :)
(and something we can play with later)

Yeah, let's just go to seconds.